xbmc/network/GUIDialogNetworkSetup.cpp

```diff
@@ -371,90 +357,51 @@ void CGUIDialogNetworkSetup::UpdateButtons()
-                                                                        m_protocol == NET_PROTOCOL_SFTP));
+    CONTROL_ENABLE_ON_CONDITION(browseControlID,
+                                m_protocols[m_protocol].browseable);
   }
 }
 
 std::string CGUIDialogNetworkSetup::ConstructPath() const
 {
   CURL url;
-  if (m_protocol == NET_PROTOCOL_SMB)
-    url.SetProtocol("smb");
-  else if (m_protocol == NET_PROTOCOL_FTP)
-    url.SetProtocol("ftp");
-  else if (m_protocol == NET_PROTOCOL_HTTP)
-    url.SetProtocol("http");
-  else if (m_protocol == NET_PROTOCOL_HTTPS)
-    url.SetProtocol("https");
-  else if (m_protocol == NET_PROTOCOL_DAV)
-    url.SetProtocol("dav");
-  else if (m_protocol == NET_PROTOCOL_DAVS)
-    url.SetProtocol("davs");
-  else if (m_protocol == NET_PROTOCOL_UPNP)
-    url.SetProtocol("upnp");
-  else if (m_protocol == NET_PROTOCOL_RSS)
-    url.SetProtocol("rss");
-  else if (m_protocol == NET_PROTOCOL_NFS)
-    url.SetProtocol("nfs");
-  else if (m_protocol == NET_PROTOCOL_SFTP)
-    url.SetProtocol("sftp");
-    
+  url.SetProtocol(m_protocols[m_protocol].type);
+
   if (!m_username.empty())
   {
     url.SetUserName(m_username);
     if (!m_password.empty())
       url.SetPassword(m_password);
   }
-  if(!m_server.empty())
+
+  if (!m_server.empty())
     url.SetHostName(m_server);
-  if (((m_protocol == NET_PROTOCOL_FTP) ||
-       (m_protocol == NET_PROTOCOL_HTTP) ||
-       (m_protocol == NET_PROTOCOL_HTTPS) ||
-       (m_protocol == NET_PROTOCOL_DAV) ||
-       (m_protocol == NET_PROTOCOL_DAVS) ||
-       (m_protocol == NET_PROTOCOL_RSS) ||
-       (m_protocol == NET_PROTOCOL_SFTP) ||
-       (m_protocol == NET_PROTOCOL_NFS))
-      && !m_port.empty() && atoi(m_port.c_str()) > 0)
+
+  if (m_protocols[m_protocol].port && !m_port.empty()
+                                   && atoi(m_port.c_str()) > 0)
   {
     url.SetPort(atoi(m_port.c_str()));
   }
+
   if (!m_path.empty())
     url.SetFileName(m_path);
+
   return url.Get();
 }
 
 void CGUIDialogNetworkSetup::SetPath(const std::string &path)
 {
   CURL url(path);
-  if (url.IsProtocol("smb"))
-    m_protocol = NET_PROTOCOL_SMB;
-  else if (url.IsProtocol("ftp"))
-    m_protocol = NET_PROTOCOL_FTP;
-  else if (url.IsProtocol("http"))
-    m_protocol = NET_PROTOCOL_HTTP;
-  else if (url.IsProtocol("https"))
-    m_protocol = NET_PROTOCOL_HTTPS;
-  else if (url.IsProtocol("dav"))
-    m_protocol = NET_PROTOCOL_DAV;
-  else if (url.IsProtocol("davs"))
-    m_protocol = NET_PROTOCOL_DAVS;
-  else if (url.IsProtocol("upnp"))
-    m_protocol = NET_PROTOCOL_UPNP;
-  else if (url.IsProtocol("rss"))
-    m_protocol = NET_PROTOCOL_RSS;
-  else if (url.IsProtocol("nfs"))
-    m_protocol = NET_PROTOCOL_NFS;
-  else if (url.IsProtocol("sftp") || url.IsProtocol("ssh"))
-    m_protocol = NET_PROTOCOL_SFTP;
-  else
-    m_protocol = NET_PROTOCOL_SMB;  // default to smb
+  m_protocol = 0;
```

Comment:
the old code blindly used smb no matter as default value, which would point to .. nothing if smb was not enabled. if you want to old behavior of the spinner showing label 0 = 'Programs' i can add it but that doesn't sound very productive ;)

Reply:
Okay only want to know if that is desired.

Reply:
i don't know what would be a better approach - maintaining a bug certainly won't be. i can look for some protocol that is always there, but that would give perceived differences when smb is enabled.

Reply:
Calling this method with a path that does not contain a valid protocol should throw an error or shouldn't set the path. Before and now it suppose the first is the right one, but that's not the case.

Reply:
no path set, made function bool and return to caller. but there is no sane way to do exceptions here and it would be rather alien to other code in kodi.

Reply:
okay cool thanks. Instead of just log an error should we inform the user about the issue (ok dialog + user-friendly message)?

Reply:
not sure how friendly i want to be to peeps who have messed up the xml by hand ;)

Reply:
okay leave it as it is, it's definitely better as before. IMO we can merge if all is fine.

Reply:
dialog added. sorry whomever have string additions in the queue.

Reply:
I've moved all my strings in my media importing work to 40000+ ;-)